src/apps/frontend/pages/tasks/task-header.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import toast from 'react-hot-toast';
+
+import {
+  Button, Flex, FlexItem, HeadingLarge,
+} from '../../components';
+import { AsyncError } from '../../types';
+import { TaskOperationType } from '../../types/task';
+
+import TaskCard from './task-card';
+import TaskModal from './task-modal';
+import useTaskForm from './tasks-form.hook';
+
+interface TaskHeaderProps {
+  onError?: (error: AsyncError) => void;
+}
+
+const TaskHeader: React.FC<TaskHeaderProps> = ({ onError }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const onSuccess = () => {
+    toast.success('Task has been added successfully');
+    setIsModalOpen(false);
+  };
+
+  const { addTaskFormik, setFormikFieldValue } = useTaskForm({
+    onError,
+    onSuccess,
+  });
+
+  return (
+    <TaskCard>
```

Comment:
This is not expected here. A TaskCard is essentially tied to a task, but here you are showing header, CTA etc. This seems like a poor abstraction.